app/services/html_parser.rb

```diff
@@ -1,16 +1,37 @@
-module CodeBlockParser
+class HtmlParser
+  # Each of the instance methods should return self to support chaining of
+  # methods
+  # For example:
+  #  HtmlParser.
+  #    new(html).
+  #    remove_nested_linebreak_in_list.
+  #    prefix_all_images.
+  #    wrap_all_images_in_links.
+  #    html
+
   include InlineSvg::ActionView::Helpers
   include ApplicationHelper
 
-  def remove_nested_linebreak_in_list(html)
-    html_doc = Nokogiri::HTML(html)
+  RAW_TAG_DELIMITERS = ["{", "}", "raw", "endraw", "----"].freeze
+
+  attr_reader :html
+
+  def initialize(html)
+    @html = html
+  end
+
+  def remove_nested_linebreak_in_list
+    html_doc = Nokogiri::HTML(@html)
```

Comment:
this is just personal preference but by having an attribute reader defined, I'd use that in lieu of the instance variable `@html`